src/SFML/Audio/InputSoundFile.cpp

```diff
@@ -73,13 +73,14 @@ std::optional<InputSoundFile> InputSoundFile::openFromFile(const std::filesystem
     if (!reader)
         return std::nullopt;
 
-    // Wrap the file into a stream
-    auto file = std::make_unique<FileInputStream>();
-
-    // Open it
-    if (!file->open(filename))
+    // Open the file
+    auto fileInputStream = FileInputStream::open(filename);
+    if (!fileInputStream)
         return std::nullopt;
 
+    // Wrap the file into a stream
+    auto file = std::make_unique<FileInputStream>(std::move(*fileInputStream));
```

Comment:
This was a little awkward. We start with a fallible factory function, check if it succeeded, and then move that file input stream into the unique pointer. I'm not aware of a more graceful way to do this.

Reply:
Hmm, I think the way you did it is fine. 

I think the real issue here is that we're unnecessarily using a unique ptr, a `std::variant<FileInputStream, MemoryInputStream, InputStream*>` would suffice. Maybe I'll send a PR for that

Reply:
I briefly tried the idea above but it's not worth doing, the reader relies on the stream's address being stable.